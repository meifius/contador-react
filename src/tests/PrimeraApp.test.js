// Importar Bibliotecas
import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import PrimeraApp from '../PrimeraApp';

describe('Testeos sobre el componente <PrimeraApp />', () => {

  // test('Debe mostrar el mensaje "Hola, soy Goku"', () => {
  //     const saludo = 'Hola, soy Goku';
  //     // Producto renderizado
  //     const {getByText} = render( <PrimeraApp saludo={saludo}/> );
  //     expect(getByText(saludo)).toBeInTheDocument();
  // })
  
  test('Debe mostrar <PrimeraApp /> correctamente', () => {
    const saludo = 'Hola, soy Goku';
    const wrapper = shallow( <PrimeraApp saludo={saludo}/> );

    expect( wrapper ).toMatchSnapshot();
  });

  test('Debe de mostrar el subtitulo enviado por props', () => {
    const saludo = 'Hola, soy Goku';
    const subTitulo = 'Soy un subtitulo';
    const wrapper = shallow( <PrimeraApp saludo={saludo} subtitulo={subTitulo}/> );

    const textoParrafo = wrapper.find('p').text();

    expect(textoParrafo).toBe(subTitulo);
  });
  
    

});