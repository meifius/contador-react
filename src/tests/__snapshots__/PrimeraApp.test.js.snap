// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeos sobre el componente <PrimeraApp /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, soy Goku
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
